### PR TITLE
Improve CI

### PR DESCRIPTION
**Pull request by VIKTORVAV99**
This PR aims to improve and consolidate the CI we already have.

In this PR:
- Adds a ESLint job.
- Makes all jobs run on a workflow file change.
- Brings over the geometry verification job from circleci.
- Brings over the jsonlint job from circleci.
- Adds a python formatting job to check python formatting with black and isort (part of the python check job from circleci).
- Enable Dependabot for GitHub Actions (ONLY).

Logical CI layout in this PR:
image

Future improvement suggestions:
- More aggressive caching (primarily the node_modules).
- Bring over the python tests.
- Bring over the frontend build.

### CODE REVIEW

**Comment by VIKTORVAV99** [comment]
@madsnedergaard I know you are on the fence about using Dependabot (for now) but I have only enabeled it for GitHub Actions and with the changes in this PR it should be more or less self evaluating (i.e. either they pass or fail).

I expect it to create a few PR:s as soon as this is merged (or the day after) to bring everything up to date but after that it should be relatively quiet (most actions don't update very often).

I'd be more than happy to sort out the immediate PR:s by Dependabot myself.

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -30,24 +35,75 @@ jobs:
...
+    needs: [setup, prettier]
+    if: needs.setup.outputs.frontend != 'false' || needs.setup.outputs.workflows != 'false'
+    uses: ./.github/workflows/verify.yml
+
   cypress:
     name: Cypress
-    needs: [setup]
-    if: needs.setup.outputs.frontend != 'false'
+    needs: [eslint, verify]
+    if: needs.setup.outputs.frontend != 'false' || needs.setup.outputs.workflows != 'false'
```
[the remark this answers is not in the document]
Comment (VIKTORVAV99) [reply .github/workflows/CI.yml:100]: Good catch, I think it does but we can remove the if statement all together as it depends on eslint and eslint has the same conditions.

**Comment by VIKTORVAV99** [comment]
> What are your thoughts on "picking" GH Actions over CircleCI? I think it makes sense to unify and only use one, just curious to hear more :)

I also think it makes sense to unify them and I would personally prefer to use GitHub actions. It would allow us to unify more of the workflow in one place (GitHub) but I also think it would be easier to maintain in the future with the help of GitHub Actions (for example the cypress action) and Dependabot but functionality wise it shouldn't be that different.

**Review by madsnedergaard — APPROVED** [review]
That makes sense, I was mostly curious if there's any features from CircleCI that we would lose (except for pricing), but I think this move is the best way to consolidate :)

**Comment by VIKTORVAV99** [comment]
> That makes sense, I was mostly curious if there's any features from CircleCI that we would lose (except for pricing), but I think this move is the best way to consolidate :)

The only feature I'm not sure about is Docker and how/if layer caching works on Github Actions. (the reason I skipped the build step in this PR).

I'll look into it but in either case it should just be temporary until we can get rid of that nodejs server.

**Pull request merged**